contrib/sarplus/python/pysarplus/SARPlus.py

```diff
@@ -0,0 +1,362 @@
+"""
+This is the one and only (to rule them all) implementation of SAR.
+"""
+
+import logging
+import pyspark.sql.functions as F
+import pandas as pd
+from pyspark.sql.types import StringType, DoubleType, StructType, StructField, IntegerType, FloatType
+from pyspark.sql.functions import pandas_udf, PandasUDFType
+from pysarplus import SARModel
+
+SIM_COOCCUR = "cooccurrence"
+SIM_JACCARD = "jaccard"
+SIM_LIFT = "lift"
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger('sarplus')
+
+class SARPlus:
+    """SAR implementation for PySpark"""
+
+    def __init__(
+        self,
+        spark,
+        col_user="userID",
+        col_item="itemID",
+        col_rating="rating",
+        col_timestamp="timestamp",
+        table_prefix="",
```

Comment:
`table_prefix` cannot pre-exist in the caching point, so do we want to add any detection scheme in the codes for that? So far, in my production system I have something manual for this. The problem comes from the use case when I want to have multiple SAR+ models created on the same cluster, hence the same caching point, at the same time, and I need to make sure that the prefixes are unique for the models.

Reply:
I'd suggest to use table_prefix=str(uuid.uuid4())? Does that work?